imap/mboxlist.c

```diff
@@ -1063,8 +1063,12 @@ static int mboxlist_create_namecheck(const char *mboxname,
         /* XXX should isadmin override this? */
         if (!config_getswitch(IMAPOPT_ALLOWJUNKTRASHSUBFOLDERS)) {
             struct buf attrib = BUF_INITIALIZER;
+            mbname_t *mbname;
 
-            annotatemore_lookup(mbentry->name, "/specialuse", userid, &attrib);
+            mbname = mbname_from_intname(mbentry->name);
```

Comment:
nit: why not declare mbname_t *mbname here, rather than two lines above?

Reply:
Because I like to keep my variable declarations at the top of the block, so I know where to look for them, and I have a nice neat list of things to make sure are cleaned up at the end of the scope.  The C++/C99 style of scattering them around just wherever is very annoying and adds a lot of mental effort to memory management.